apps/judicial-system/backend/src/app/modules/subpoena/subpoena.service.ts

```diff
@@ -119,36 +176,36 @@ export class SubpoenaService {
       requestedDefenderName,
     } = update
 
-    const notificationType = isSuccessfulServiceStatus(serviceStatus)
-      ? NotificationType.SERVICE_SUCCESSFUL
-      : isFailedServiceStatus(serviceStatus)
-      ? NotificationType.SERVICE_FAILED
-      : defenderChoice === DefenderChoice.CHOOSE && defenderNationalId
-      ? NotificationType.DEFENDANT_SELECTED_DEFENDER
-      : undefined
-
     const [numberOfAffectedRows] = await this.subpoenaModel.update(update, {
       where: { subpoenaId: subpoena.subpoenaId },
       returning: true,
       transaction,
     })
-    let defenderAffectedRows = 0
 
-    // If there is a change in the defender choice after the judge has confirmed the choice,
-    // we need to set the isDefenderChoiceConfirmed to false
-    const isChangingDefenderChoice =
-      (update.defenderChoice &&
-        subpoena.defendant?.defenderChoice !== update.defenderChoice) ||
-      (update.defenderNationalId &&
-        subpoena.defendant?.defenderNationalId !== update.defenderNationalId &&
-        subpoena.defendant?.isDefenderChoiceConfirmed)
+    if (numberOfAffectedRows !== 1) {
```

Comment:
It could be normal for the number of affected rows to be 0 if all the update is doing is updating the defenderChoice for example, since that field is in the defendant table

Reply:
Interesting, had not thought of that.

Reply:
I changed this and the following check for defendant. Later I think it would be better to not call `update` unless there is something to update and then throw an exception if no rows are affected.